evm_arithmetization/src/generation/jumpdest.rs

```diff
@@ -0,0 +1,191 @@
+use std::cmp::max;
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    ops::{Deref, DerefMut},
+};
+
+use itertools::{sorted, Itertools};
+use keccak_hash::H256;
+use serde::{Deserialize, Serialize};
+
+/// Each `CodeAddress` can be called one or more times, each time creating a new
+/// `Context`. Each `Context` will contain one or more offsets of `JUMPDEST`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ContextJumpDests(pub HashMap<usize, BTreeSet<usize>>);
+
+/// The result after proving a `JumpDestTableWitness`.
```

Comment:
```suggestion
/// The result after proving a [`JumpDestTableWitness`].
```
nit: hyperlink so if it get stale some build should fail

(can someone check we do `cargo doc -- -D warnings` somewhere in CI?)

Reply:
> (can someone check we do cargo doc -- -D warnings somewhere in CI?)

```
rg "doc[^k]" .github/
.github/labeler.yml
21:# Add 'specs' label to any changes within 'docs' or `book` folder.
24:  - any-glob-to-any-file: ['docs/**', 'book/**']

.github/workflows/ci.yml
180:    name: Rustdoc, Formatting and Clippy
207:        run: cargo doc --all

```
So we don't, but it is also an invalid command:

```
cargo doc -- -D warnings
error: unexpected argument '-D' found

Usage: cargo doc [OPTIONS]

For more information, try '--help'.

```

For the record: `--all` is deprecated in favour of `--workspace`.

Reply:
`RUSTDOCFLAGS="-D warnings" cargo doc --workspace` (or `--all`) fails in my local repo.

Reply:
Have you passed `--no-deps`? Otherwise you're calling `cargo doc` on every dependency, and failing on any warning. Otherwise it seems you had an issue with your rebase, the highlighted error is fixed on `develop`.

<img width="732" alt="Screenshot 2024-10-02 at 10 53 24 AM" src="https://github.com/user-attachments/assets/e5dbe8b9-12a4-4a6f-86cd-773b7481ba1a">

Reply:
my bad, passes on `develop`.